src/build.cc

```diff
@@ -388,7 +388,10 @@ void Plan::EdgeFinished(Edge* edge) {
   edge->outputs_ready_ = true;
 
   // See if this job frees up any delayed jobs
-  ResumeDelayedJobs(edge);
+  if (directly_wanted)
+    ResumeDelayedJobs(edge);
```

Comment:
ResumeDelayedJobs is only two lines and only called here. If you inline the function, this becomes

  if (directly_wanted)
    edge->pool()->EdgeFinished(*edge);
  edge->pool()->RetrieveReadyEdges(&ready_);

And directly_wanted is identical to e->second in this scope, right? …oh, I guess it isn't as we recurse down.

If we recurse, do we want the state of where we started recursing to decide if we should call EdgeFinished(), or the local state of that edge? Probably the latter, right? So not passing in a bool is the more correct version too as far as I understand?